File: Release/WetiMatlabFunctions/FileOperations/AddLineToTXTFile.m
% Function: AddLineToTXTFile add lines to .txt file
% -----------------------------
% Usage:
% AddLineToTXTFile(TXTFile,nLine,NewLine)
% -----------------------------
% Input:
% TXTFile           String with name of .txt file
% nLine             line after which the new line is added
% NewLine           String with new line
% -----------------------------
function AddLineToTXTFile(TXTFile,nLine,NewLine)

[FOLDER,NAME,EXT]   = fileparts(TXTFile); 
TempTXTFile         = fullfile(FOLDER,[NAME,'_temp',EXT]);
fid                 = fopen(TXTFile);
fidTemp             = fopen(TempTXTFile,'w+');

% copy file up to nLine
for iLine=1:nLine
   s            = fgetl(fid);
   fprintf(fidTemp,'%s\r\n',s);
end

% add new line
fprintf(fidTemp,'%s\r\n',NewLine);

% copy rest of file
while ~feof(fid)
   s            = fgetl(fid);
   fprintf(fidTemp,'%s\r\n',s);
end

% close files
fclose(fid);
fclose(fidTemp);
delete(TXTFile);
movefile(TempTXTFile,TXTFile);

File: Release/PythonFunctions/FileOperations/AddLineToTXTFile.py
import os

def AddLineToTXTFile(txt_file, n_line, new_line):
    folder, name = os.path.split(txt_file)
    name, ext = os.path.splitext(name)
    temp_txt_file = os.path.join(folder, f'{name}_temp{ext}')

    with open(txt_file, 'r') as fid, open(temp_txt_file, 'w+') as fid_temp:
        # copy file up to n_line
        for i_line in range(n_line):
            s = fid.readline()
            fid_temp.write(s)

        # add new line
        fid_temp.write(new_line + '\n')

        # copy rest of file
        for s in fid:
            fid_temp.write(s)

    # delete original file and rename temp file
    os.remove(txt_file)
    os.rename(temp_txt_file, txt_file)
